resources/js/components/data-list/BulkActions.vue

```diff
@@ -42,56 +43,52 @@ export default {
         }
     },
 
-    computed: {
+    data() {
+        return {
+            actions: [],
+        };
+    },
 
+    computed: {
         selections() {
             return this.sharedState.selections;
         },
 
         hasSelections() {
             return this.selections.length > 0;
         },
+    },
 
-        actions() {
-            const rows = this.sharedState.rows.filter(row => this.selections.includes(row.id));
-
-            let actions = rows.reduce((carry, row) => carry.concat(row.actions), []);
-
-            actions = _.uniq(actions, 'handle').filter(action => action.bulk);
-
-            // Remove any actions that are missing from any row. If you can't apply the action
-            // to all of the selected items, you should not see the button. There's server
-            // side authorization for when the action is executed anyway, just in case.
-            rows.forEach(row => {
-                actions = actions.filter(action => row.actions.map(a => a.handle).includes(action.handle));
-            });
-
-            return _.sortBy(actions, 'title');
-        }
-
+    watch: {
+        selections: 'getActions',
```

Comment:
I didn't know you could do this. Neat!

Reply:
Yes, we're [doing it here already](https://github.com/statamic/cms/blob/master/resources/js/components/data-list/FieldFilter.vue#L127) 👍